### PR TITLE
Fix ConstraintDualStart with VariableIndex

### LINKED ISSUE #333: copy_to does not copy attributes for variable bound constraints
When calling copy_to, at some point MOI.pass_attributes gets called (via MOI.Utilities.default_copy_to()):
```julia
function MOI.copy_to(model::Optimizer, src::MOI.ModelLike)
    return MOI.Utilities.default_copy_to(model, src)
end
```
```julia
function pass_attributes(
    dest::MOI.ModelLike,
    src::MOI.ModelLike,
    index_map::IndexMap,
    cis_src::Vector{MOI.ConstraintIndex{F,S}},
) where {F,S}
    for attr in MOI.get(src, MOI.ListOfConstraintAttributesSet{F,S}())
        if !MOI.supports(dest, attr, MOI.ConstraintIndex{F,S})
            if (
                attr == MOI.ConstraintName() ||
                attr == MOI.ConstraintPrimalStart() ||
                attr == MOI.ConstraintDualStart()
            )
                continue  # Skipping names and start values is okay.
            end
        end
        _pass_attribute(dest, src, index_map, cis_src, attr)
    end
    return
end
```
currently, MOI.supports() in this case is defined as follows: 
```julia
function MOI.supports(
    ::Optimizer,
    ::MOI.ConstraintDualStart,
    ::Type{MOI.ConstraintIndex{F,S}},
) where {F<:_FUNCTIONS,S<:_SETS}
    return true
end
```
where 
```julia
const _SETS =
    Union{MOI.GreaterThan{Float64},MOI.LessThan{Float64},MOI.EqualTo{Float64}}

const _FUNCTIONS = Union{
    MOI.ScalarAffineFunction{Float64},
    MOI.ScalarQuadraticFunction{Float64},
}
```
This results in the fact that attributes for variable bound constraints are not copied, since calling MOI.supports() for bound constraints (where F=MOI.VariableIndex) returns the default false and attributes are not passed using
```julia
 _pass_attribute(dest, src, index_map, cis_src, attr)
```
Extending the MOI.supports definition to variable bounds creates a lot of other problems. 
However, things such as ConstraintDualStart have get/set methods that work as desired. 
Would it be a good idea to include these attributes manually in the copy_to function? I was thinking of something like this: 
```julia
function MOI.copy_to(model::Optimizer, src::MOI.ModelLike)
    idxmap =  MOI.Utilities.default_copy_to(model, src)
    # copy relevant attributes here
    return idxmap
end
```
If so, I would be happy to start with a PR at least implementing the dual start values. 
In case this is not a good idea, a brief warning about not copying bound dual starting values would maybe be good. 
This also relates to the other attributes skipped by the continue statement in pass_attributes(), but I think bound dual starting values are the most relevant (primal starts are defined by variable primal starts, and copying names for bounds is also not really relevant...).

### PR DESCRIPTION
**Pull request by odow**
Closes #333

### CODE REVIEW

**Review by blegat — APPROVED** [review]

File: test/MOI_wrapper.jl [hunk 1]
```diff
@@ -48,6 +48,8 @@ function test_MOI_Test()
             ],
         );
         exclude = String[
+            # TODO(odow): this failure is unexpected.
+            "test_model_ModelFilter_AbstractConstraintAttribute",
```
Comment (odow) [inline test/MOI_wrapper.jl:52]: This is actually a problem. We shouldn't merge this PR just yet:
```Julia
julia> import MathOptInterface

julia> import Ipopt

julia> const MOI = MathOptInterface
MathOptInterface

julia> model = MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}())
MOIU.UniversalFallback{MOIU.Model{Float64}}
fallback for MOIU.Model{Float64}

julia> x = MOI.add_variable(model)
MathOptInterface.VariableIndex(1)

julia> c = MOI.add_constraint(model, x, MOI.GreaterThan(0.0))
MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.GreaterThan{Float64}}(1)

julia> MOI.set(model, MOI.ConstraintDualStart(), c, 2.0)

julia> solver = MathOptInterface.instantiate(
           Ipopt.Optimizer;
           with_bridge_type = Float64,
       )
MOIB.LazyBridgeOptimizer{Ipopt.Optimizer}
with 0 variable bridges
with 0 constraint bridges
with 0 objective bridges
with inner model Ipopt.Optimizer

julia> index_map = MOI.copy_to(solver, model)
ERROR: BoundsError: attempt to access 0-element Vector{Int64} at index [0]
Stacktrace:
  [1] getindex
    @ ./array.jl:805 [inlined]
  [2] bridge_index
    @ ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/graph.jl:290 [inlined]
  [3] bridge_index
    @ ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/lazy_bridge_optimizer.jl:426 [inlined]
  [4] bridge_type(b::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, F::Type{MathOptInterface.VariableIndex}, S::Type{MathOptInterface.GreaterThan{Float64}})
    @ MathOptInterface.Bridges ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/lazy_bridge_optimizer.jl:491
  [5] concrete_bridge_type(b::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, F::Type{MathOptInterface.VariableIndex}, S::Type{MathOptInterface.GreaterThan{Float64}})
    @ MathOptInterface.Bridges.Constraint ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/Constraint/bridge.jl:86
  [6] supports(b::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, attr::MathOptInterface.ConstraintDualStart, IndexType::Type{MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.GreaterThan{Float64}}})
    @ MathOptInterface.Bridges ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/bridge_optimizer.jl:1369
  [7] pass_attributes(dest::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, index_map::MathOptInterface.Utilities.IndexMap, cis_src::Vector{MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.GreaterThan{Float64}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/MMUy6/src/Utilities/copy.jl:116
  [8] _pass_constraints(dest::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, index_map::MathOptInterface.Utilities.IndexMap, variable_constraints_not_added::Vector{Any})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/MMUy6/src/Utilities/copy.jl:330
  [9] default_copy_to(dest::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/MMUy6/src/Utilities/copy.jl:498
 [10] copy_to(dest::MathOptInterface.Bridges.LazyBridgeOptimizer{Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}})
    @ MathOptInterface.Bridges ~/.julia/packages/MathOptInterface/MMUy6/src/Bridges/bridge_optimizer.jl:451
 [11] top-level scope
    @ REPL[37]:1
```

**Comment by odow** [comment]
This PR exposed a bug in MOI: https://github.com/jump-dev/MathOptInterface.jl/pull/1991. So I'll wait until we have a v1.8.1 release before merging this.

**Comment by odow** [comment]
Needs a v1.8.1 release of MOI: https://github.com/jump-dev/MathOptInterface.jl/pull/1994

**Comment by odow** [comment]
Fixed!

**Pull request merged**